### PR TITLE
.equals() should compare dataValues directly

### PR DESCRIPTION
**Pull request by MrVoltz**
Avoids changes created by other's getters

### CODE REVIEW

**Comment by mickhansen** [comment]
How are you facing this problem? getters are ideally pure and side-effect free so for two instances with the same values the getters should return the same values aswell.

**Comment by MrVoltz** [comment]
You are comparing non-transformed value with transformed value, so they can be different. You need to either compare transformed with transformed or non-transformed with non-transformed. I patched the function to compare non-transformed with non-transformed.

**Comment by mickhansen** [comment]
It's doing a key lookup on both, so it should be comparing between transformed values.

File: lib/instance.js [hunk 1]
```diff
@@ -993,12 +993,15 @@ Instance.prototype.decrement = function(fields, options) {
  */
 Instance.prototype.equals = function(other) {
   var result = true;
+  
+  if(!other || !other.dataValues)
+    return false;
```
Comment (mickhansen) [inline lib/instance.js]: Style: Either isngle line or with brackets, also needs a space after the if.

**Pull request merged**